src/display.rs

```diff
@@ -207,3 +209,155 @@ mod tests {
                     \"foo.bar\" = 2\n");
     }
 }
+
+
+
+
+/// Encodes an encodable value into a "pretty" TOML string.
+///
+/// This function expects the type given to represent a TOML table in some form.
+/// If encoding encounters an error, then this function will fail the task.
+/// 
+/// "pretty" means the following features of the output are changed:
+/// - strings with newlines characters (`\n`) will use the `'''` form
+///     and span multiple lines
+pub fn encode_str_pretty<'a>(tbl: TomlTable) -> Result<StdString, fmt::Error> {
```

Comment:
Ideally this would look like:

```rust
impl Value {
    pub fn pretty(&self) -> Pretty {
        // ...
    }
}

pub struct Pretty<'a> {
    inner: &'a Value,
}

impl<'a> fmt::Display for Pretty<'a> {
    // ...
}
```

That way you could just do:

```rust
let s = val.pretty().to_string();
```

or

```rust
let s = format!("toml: {}", val.pretty());
```